File: python/examples/submit_response.py
#!/usr/bin/env python

import os
import sys
sys.path.insert(0, os.path.join(os.path.dirname(
    os.path.dirname(os.path.abspath(__file__))), 'src'))

from datetime import datetime

from epidb_client import EpiDBClient

import config

api_key = config.api_key
data = {
    'user_id': config.user_id,
    'date': datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
    'survey_id': config.user_id,
    'answers': {
        'q1': 1,
        'q2': True,
        'q3': [ 1, 2, 3 ],
        'q4': 'Jakarta'
    }
}

client = EpiDBClient(api_key)
result = client.response_submit(data)

status = result['stat']

print "status:", status

if status == 'ok':
    print "id:", result['id']
else:
    print "error code:", result['code']
    print "       msg:", result['msg']

File: php/examples/submit_response.php
<?

require_once('../EpiDBClient.php');
require('config.php');

$api_key = $_api_key;
$user_id = $_user_id;
$survey_id = $_survey_id;
$answers = array('q1' => 1,
                 'q2' => True,
                 'q3' => array( 1, 2, 3 ),
                 'q4' => 'Jakarta');

// If $date is null, the current date and time will be used
//   $date = null;
$date = '2009-12-15 01:02:03';

$client = new EpiDBClient($api_key);
$client->server = $_server; // Use this if you want to override
                            // the server location
$result = $client->response_submit($user_id, $survey_id, $answers, $date);

$status = $result['stat'];

header('Content-Type: text/plain');

print("status: " . $status . "\n");

if ($status == 'ok') {
    print("id: " . $result['id'] . "\n");
}
else {
    print("error code: " . $result['code'] . "\n");
    print("       msg: " . $result['msg'] . "\n");
}

